tokio/src/runtime/builder.rs

```diff
@@ -66,6 +66,9 @@ pub struct Builder {
 
     /// To run before each worker thread stops
     pub(super) before_stop: Option<Callback>,
+
+    #[cfg(all(feature = "syscall", tokio_unstable))]
+    pub(super) syscalls: Option<Box<dyn crate::syscall::Syscalls>>,
```

Comment:
Is there a reason we have `syscalls`in a `Box` here and then convert it into an `Arc` when building the runtime? Taking an `Arc` would make it easier to share one `Syscalls` instance between multiple runtimes, which I can see being useful in at least some testing use-cases (e.g. simulating multiple components of a distributed system in process and using a shared `Syscalls` instance to simulate network partitions or congestion). Is there a reason we want to disallow sharing a `Syscalls` instance across runtimes?

Reply:
I was thinking only about the simulation testing use case here, where you don't want to share the same `Syscalls` instance across multiple runtime. I had not considered that a user may want to write tests using multiple runtimes, but that sounds useful. I've switched the `Runtime` `Builder` to take an `Arc` instead. Thanks!